plugins/generator-1.18.2/forge-1.18.2/templates/rangeditem/projectile_entity.java.ftl

```diff
@@ -136,8 +136,10 @@ public class ${name}Entity extends AbstractArrow implements ItemSupplier {
 			}/>
 		</#if>
 
+        <#if !data.preserveProjectiles>

```

Comment:
I'd also add a check for `hasProcedure(data.onBulletFlyingTick) || !data.preserveProjectiles` around the entire tick method, so that we don't have an unnecessary override